libs/libarchfpga/src/grid_types.h

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+/**
+ * @file grid_types.h
+ * @brief FPGA grid layout data types 
+ */
+
+#include <string>
+#include <memory>
+
+// Forward declerations
+
+struct t_metadata_dict;
+
+/**
+ * @brief Grid location specification
+ * Each member is a formula evaluated in terms of 'W' (device width),
+ * and 'H' (device height). Formulas can be evaluated using parse_formula()
+ * from expr_eval.h.
+ */
+struct t_grid_loc_spec {
```

Comment:
I think we should build these into the grid part of the VPR API documentation on the web. You'd need to edit some doxyfile for that, and add an overview at the top of this struct saying this is used to capture the user's specification of the architecture will eventually be turned into a flattened grid.